lib/functions/Core.ts

```diff
@@ -12,21 +8,69 @@ type Term = E.TermExpression;
 // Overloaded Functions
 // ----------------------------------------------------------------------------
 
-// Maps argument types on their specific implementation.
-export type OverloadMap = Map<List<ArgumentType>, E.SimpleApplication>;
+type SearchStack = OverloadNode[];
+
+export class OverloadNode {
```

Comment:
Let's move this to a separate file.

Reply:
Also, since it is not trivial to understand what is going on in this class, I would suggest adding brief jsdoc to explain why it exists, and what it does.

Reply:
Alright!